extensions/amp-form/0.1/amp-form.js

```diff
@@ -902,14 +909,37 @@ export class AmpForm {
     ).then(request => this.xhr_.fetch(request.xhrUrl, request.fetchOpt));
   }
 
+  /**
+   * Returns the action trust for submit-success and submit-error events.
+   * @param {!ActionTrust} incomingTrust
+   * @return {!ActionTrust}
+   * @private
+   */
+  trustForSubmitResponse_(incomingTrust) {
+    // TODO(choumx): Remove this expected error before Q1 2020.
+    if (incomingTrust <= ActionTrust.DEFAULT) {
+      dev().expectedError(
+        TAG,
+        'Recursive form submissions are intended to be deprecated. ' +
```

Comment:
Since you have a comment to remove the error, it may be helpful to include the timeline in the error also. "Recursive form submissions are intended to be deprecated by Jan 1, 2020"

Reply:
Good idea, done.